index.rst

```diff
@@ -5,77 +5,331 @@
 Abstract
 ========
 
-The First-Look Analysis and Feedback Functionality (FAFF) working group, identified the need for a service that is capable of executing user-defined operations/computation following user-defined rules.
-This service has been referred  to as Catcher.
-In essence, the Catcher is not that different from the Watcher or other stream processing off-the-shelf technologies like Kapacitor.
-This technote explores the idea of using InfluxDB OSS 2.x new task engine as an alternative implementation for the Catcher.
+This technote contains design considerations for the Catcher, an event-driven processing component for the Rubin Observatory Control System.
+We briefly review the existing functionality of the system and discuss the use case for the Catcher.
+The idea originated from the First-Look Analysis and Feedback Functionality (FAFF) working group, and a more in-depth analysis of the use cases can be found in their reports.
+
 
 Introduction
 ============
 
 One of the finding of the First-Look Analysis and Feedback Functionality (FAFF) working group :sitcomtn:`025` :cite:`SITCOMTN-025`, was the need for a service to ...
 
-   handle the low-level coordiation of identifying when a specific condition is met, then launching and monitoring an analysis process.
+   handle the low-level coordination of identifying when a specific condition is met, then launching and monitoring an analysis process.
 
-The original proposal from the FAFF working group was to develop a Commandable SAL Component (CSC), similar to the Watcher, that would provide be capable of monitoring data from the system and trigger certain actions (e.g. perform some computation and/or generate reports) based on some pre-defined conditions.
+The idea is basically to have a component capable of executing event-driven processing and report generation.
 
-Following additional discussions, we have decided to experiment with an alternative approach, replacing the CSC with an off-the-shelf solution provided by provided by Sasquatch (see :sqr:`068` :cite:`SQR-068`).
+The Rubin Observatory Control System currently contains a ``Watcher`` CSC that is capable of generating alerts from user defined rules (e.g. event-driven alerts).
+Moreover, the ``ScriptQueue`` and ``OCPS`` CSCs are both able to execute command-driven processes.
+While the ``ScriptQueue`` is tailored towards control system operations (e.g. moving the telescope, taking data, etc.), the ``OCPS`` is dedicated to executing DM-pipeline tasks processes.
+The missing piece of functionality is the capability of mixing a ``Watcher``-like behavior (event-processing) with ``ScriptQueue``/``OCPS``-like behavior.
```

Comment:
Well, I don't think that's quite all of it.
It's that we don't have the ability to act on an event, where that act is running a certain amount of processing, then (potentially) creating an artifact for observers to look at.

Reply:
Also, note that, in terms of functionality,

> running a certain amount of processing, then (potentially) creating an artifact for observers to look at. 

is what the ScriptQueue + OCPS do. So, in a sense, I think I am saying the same thing, but maybe needs some wordsmithing?